Extensions/DialogueTree/dialoguetools.js

```diff
@@ -218,9 +284,16 @@ gdjs.dialogueTree.confirmSelectOption = function() {
     this.selectedOption !== -1
   ) {
     this.commandCalls = [];
-    this.dialogueData.select(this.selectedOption);
-    this.dialogueData = this.dialogue.next().value;
-    gdjs.dialogueTree.goToNextDialogueLine();
+    try {
+      this.dialogueData.select(this.selectedOption);
+      this.dialogueData = this.dialogue.next().value;
+      gdjs.dialogueTree.goToNextDialogueLine();
+    } catch (error) {
+      console.error(
+        `An error happened when trying to access the dialogue branch!`,
```

Comment:
What could cause this error?

Reply:
Selecting an option that doesn't exists in the yarn data tree will cause bondagejs to crash the game. This is why I have added a try catch, so instead nothing happens and a warning is logged :) Ideally this should never happen. Yarn automatically creates nodes to options you make a new link to. But for flexibility it is still possible for the user to delete a node and forget to erase a link to it. I think of later on adding a warning before saving the file for such dead node links

Reply:
Thanks to this catch, the player will still be capable of trying to select another option in the game that does have a node linked to it - and the game will proceed to  work.

Reply:
Sounds good then!